torch/onnx/_internal/fx/type_utils.py

```diff
@@ -38,6 +43,23 @@ def from_torch_dtype_to_onnx_dtype_str(dtype: Union[torch.dtype, type]) -> Set[s
     return _TORCH_DTYPE_TO_COMPATIBLE_ONNX_TYPE_STRINGS[dtype]
 
 
+def from_python_type_to_onnx_attribute_type(
```

Comment:
Question for my understanding.

- Where do we convert this python object to onnx attribute? I assume eventually we need to do this conversion.
- If my assumption above is correct, Can we merge the core of the two logics? Feels like much can be reused.
- If above is achievable, we should be able to extend support for `torch.dtype`, `torch.device`, etc as part of the python types supported here?

Reply:
> Where do we convert this python object to onnx attribute? I assume eventually we need to do this conversion.

For most of attribute (int, float, bool, str), we don't need to convert them, we only need to `filter_incompatible_and_dtype_convert_kwargs` to omit useless kwargs and JitScalarType convert torch dtype to onnx dtype. The AttributeProto is written into OpSchema with the help of param_schema(), but it was never used.

Reply:
The conversion is handled by torchscript if I understand your question right